cmd/hostpaths/hostpaths.go

```diff
@@ -420,6 +433,32 @@ func mapHostPaths(ctx context.Context, pManager, vManager manager.Manager) {
 	}, time.Second*5)
 }
 
+func filterManagedPods(ctx context.Context, options *context2.VirtualClusterOptions, podList *corev1.PodList, pManager manager.Manager) error {
```

Comment:
another note, I am not sure I like updating the pod list in place. You appear to do it because `fillUpPodMapping` wants a `PodList` but it actually only needs a `[]Pod`, and should return a `map[string]*PodDetail` instead of updating a map in place that has no other use

if you could do that that'd be awesome 🙏

Reply:
Thanks!! :slightly_smiling_face:  Yeah, I can simplify the fillUpPodMapping to work on a []Pod and return a map[string]PodDetail and on the caller side just set it directly to the variable. 

Though, I think I can't avoid the deletion I feel. Do correct me if I am wrong here.
So basically, when we are dealing with a vcluster that has [multinamespace mode](https://www.vcluster.com/docs/syncer/other_resources/multi_namespace_mode) enabled, we will have additional namespaces in the hostcluster for each namespace inside the vcluster. This would ideally be a subset of the full namespace list and would grow/shrink as namespaces are added/deleted within the vcluster. My goal here is to only process the pods that belong to any of these namespaces, since otherwise the hostpath mapping would just act on all pods in all namespaces.

Reply:
something like this 
```go
matchingPods := make([]pod, 0, Len(pods))
for _, pod := range pods {
   if vclusterNamespaces[pod.Namespace] {
       matchingPods = append(matchingPods, pod)
   }
}
return matchingPods
```

Reply:
Yeah seems so, looking at the code for slices.DeleteFunc. Also, I realised that we can merge fillUpPodMapping and filterManagedPods together into a single function that just outputs the podMapping in the end. Let me know, if this looks good now.